timeline-rest/src/main/java/cz/muni/fi/timeline/rest/controller/StudyGroupController.java

```diff
@@ -10,27 +27,105 @@
 @RestController
 @RequestMapping("/groups")
 public class StudyGroupController {
-    // TODO getStudyGroup
+
+    final static Logger logger = LoggerFactory.getLogger(StudyGroupController.class);
+
+    private StudyGroupFacade studyGroupFacade;
+    private StudyGroupAssembler studyGroupAssembler;
+
+    @Autowired
+    public StudyGroupController(StudyGroupFacade groupFacade, StudyGroupAssembler studyGroupAssembler){
+        this.studyGroupFacade = groupFacade;
+        this.studyGroupAssembler = studyGroupAssembler;
+    }
+
+    // getStudyGroup
     // GET /{id}
+    @RequestMapping(value = "/{id}", method = RequestMethod.GET, produces = MediaType.APPLICATION_JSON_VALUE)
+    public final Optional<StudyGroupDTO> getStudyGroup(@PathVariable("id") long id){
+        logger.debug("rest getStudyGroup({})", id);
+        Optional<StudyGroupDTO> studyGroupDTO = studyGroupFacade.getStudyGroupWithId(id);
+        return studyGroupDTO;
+    }
 
-    // TODO removeStudyGroup
+    // removeStudyGroup
     // DELETE /{id}
+    @RequestMapping(value = "/{id}", method = RequestMethod.DELETE, produces = MediaType.APPLICATION_JSON_VALUE)
+    public final void deleteStudyGroup(@PathVariable("id") long id) throws Exception {
+        logger.debug("rest deleteStudyGroup({})", id);
+        try {
+            studyGroupFacade.removeStudyGroup(id);
+        } catch (Exception ex) {
+            throw new Exception("resource not found");
+        }
+    }
 
-    // TODO updateStudyGroup
+    // updateStudyGroup
     // PUT /{id}
+    @RequestMapping(value = "/{id}", method = RequestMethod.PUT, consumes = MediaType.APPLICATION_JSON_VALUE,
+            produces = MediaType.APPLICATION_JSON_VALUE)
+    public final Optional<StudyGroupDTO> changePrice(@PathVariable("id") long id, @RequestBody StudyGroupDTO updatedGroup) throws Exception {
```

Comment:
Nevracej Optional. Vyhoď vyjímku — ResourceNotFoundException. Přidal jsem ty výjimky do developu. Musíš si rebasnout svoji branch. To se týká i ostatních metod.

Reply:
Podívej se i na ostatní výjimky co jsem přidal a zkus je použít na vhodných místech.